### PR TITLE
Add gfx.device.copyTextureToTexture

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #17268

This PR was originally developed by @zhoujianbo0

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/gfx/webgl2/webgl2-commands.ts [hunk 1]
```diff
@@ -3460,3 +3460,48 @@ export function WebGL2CmdFuncBlitTexture (
...
+    gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, srcGPUTextrue.glTarget, srcGPUTextrue.glTexture, 0);
+    const glTexUnit = cache.glTexUnits[cache.texUnit];
+    if (glTexUnit.glTexture != dstGPUTexture.glTexture) {
+        gl.bindTexture(gl.TEXTURE_2D, dstGPUTexture.glTexture);
+        glTexUnit.glTexture = dstGPUTexture.glTexture;
+    }
+    gl.copyTexSubImage2D(gl.TEXTURE_2D, 0, dx, dy, srcX, srcY, srcW, srcH);
+    gl.bindFramebuffer(gl.FRAMEBUFFER, null);
+    cache.glFramebuffer = null;
+    gl.deleteFramebuffer(framebuffer);
```
Comment (dumganhar) [inline cocos/gfx/webgl2/webgl2-commands.ts:3506]: No need to create / delete FBO each time. Consider to re-use it.

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by shaoqiangcai** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by shaoqiangcai** [comment]
@cocos-robot run test cases

**Comment by sushanhong** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by minggo** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

File: native/cocos/renderer/gfx-gles2/GLES2Commands.cpp [hunk 2]
```diff
@@ -2981,6 +2981,41 @@ void cmdFuncGLES2BlitTexture(GLES2Device *device, GLES2GPUTexture *gpuTextureSrc
...
+    }
+    GL_CHECK(glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, srcGPU->glTarget, glResource, 0));
+
+    GLenum status;
+    GL_CHECK(status = glCheckFramebufferStatus(GL_FRAMEBUFFER));
+    CC_ASSERT_EQ(status, GL_FRAMEBUFFER_COMPLETE);
+
+    GL_CHECK(glBindTexture(dstGPU->glTarget, dstGPU->glTexture));
+    cache->glTextures[cache->texUint] = dstGPU->glTexture;
+    GL_CHECK(glCopyTexSubImage2D(dstGPU->glTarget, 0, dstX, dstY, x, y, w, h));
```
Comment (dumganhar) [inline native/cocos/renderer/gfx-gles2/GLES2Commands.cpp:3016]: Memoryleak here. need to invoke glDeleteFramebuffers and reset cache->glReadFramebuffer to 0.

**Comment by shaoqiangcai** [comment]
@cocos-robot run test cases

**Pull request closed without merging**